libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinPlugin.kt

```diff
@@ -433,25 +437,24 @@ internal abstract class AbstractKotlinPlugin(
         }
 
         private fun configureClassInspectionForIC(project: Project) {
-            val classesTask = project.tasks.findByName(JavaPlugin.CLASSES_TASK_NAME)
-            val jarTask = project.tasks.findByName(JavaPlugin.JAR_TASK_NAME)
+            val classesTask = project.locateTaskByName(JavaPlugin.CLASSES_TASK_NAME)
+            val jarTask = project.locateTask<Jar>(JavaPlugin.JAR_TASK_NAME)
 
-            if (classesTask == null || jarTask !is Jar) {
+            if (classesTask == null || jarTask == null) {
                 project.logger.info(
-                    "Could not configure class inspection task " +
-                            "(classes task = ${classesTask?.javaClass?.canonicalName}, " +
-                            "jar task = ${classesTask?.javaClass?.canonicalName}"
+                    "Could not configure class inspection task because at least one of the 2 following tasks " +
+                            "have not been found: ${JavaPlugin.CLASSES_TASK_NAME} & ${JavaPlugin.JAR_TASK_NAME}."
                 )
                 return
             }
             val inspectTask =
-                registerTask(project, "inspectClassesForKotlinIC", InspectClassesForMultiModuleIC::class.java) {
+                project.registerTask<InspectClassesForMultiModuleIC>("inspectClassesForKotlinIC") {
                     it.sourceSetName = SourceSet.MAIN_SOURCE_SET_NAME
-                    it.archivePath.set(project.provider { jarTask.archivePathCompatible.canonicalPath })
-                    it.archiveName.set(project.provider { jarTask.archiveNameCompatible })
+                    it.archivePath.set(project.provider { jarTask.get().archivePathCompatible.canonicalPath })
+                    it.archiveName.set(project.provider { jarTask.get().archiveNameCompatible })
```

Comment:
I know it might be too big a change but here the configuration of `inspectTask` will trigger the creation of `jarTask`, but the dependency between the 2 tasks is the other way around (see line 457). Ideally it should be the `inspectTask` that would contain the data and `jarTask` can realize it during its configuration because it will know for sure that `inspectTask` will be realized. An alternative is to have the data being hold independently of tasks.